backend/classify.functions.php

```diff
@@ -0,0 +1,124 @@
+<?php
+
+// NA	North america	Split by locale between NYC3 and SFO1
+// SA	South america	Split by IP hash between NYC3 and SFO1
+
+// EU	Europe			Split by locale between AMS3, FRA1, and LON1
+// AF	Africa			Split by IP hash between AMS3 and FRA1
+
+// AS	Asia			SGP1
+// OC	Oceania			SGP1
+// AN	Antarctica		SGP1
+
+include 'geoip2.phar';
+use GeoIp2\Database\Reader;
+
+function ipCheck($hostname, $debug = false) {
+
+	try {
+		if ( $debug ) {
+			echo $hostname."\n";
+		}
+		$reader = new Reader(__DIR__.'/GeoLite2-City.mmdb');
+		$record = $reader->city($hostname);
+		if ( $debug > 1 ) {
+			var_dump($record);
+		}
+		$continent = $record->continent->code;
+		$country   = $record->country->isoCode;
+		$longitude  = $record->location->longitude;
+
+	} catch (\Exception $e) {
+		echo '<!-- '.$e->getMessage().' -->'."\n";
+		$continent = false;
+		$country   = false;
+		$longitude  = false;
+	}
+
+	if ( $debug ) {
+		echo 'Continent: "'.$continent.'"'."\n";
+		echo 'Country: "'.$country.'"'."\n";
+		echo 'Longitude: "'.$longitude.'"'."\n";
+	}
+
+	// North America
+	if ( $continent == 'NA' ) {
+		// These lists are based on who is on what side of the USA.
+		$northEast = array('BZ', 'CR', 'SV', 'GT', 'HN', 'MX', 'NI', 'PA');
+		$northWest = array('AG', 'BS', 'BB', 'BM', 'VG', 'KY', 'CU', 'DM', 'DO', 'GL', 'GD', 'GP', 'HT', 'JM', 'MQ', 'MS', 'CW', 'AW', 'SX', 'BQ', 'PR', 'BL', 'KN', 'AI', 'LC', 'MF', 'PM', 'VC', 'TT', 'TC', 'VI');
+		if (
+			in_array($country, $northEast) ||
+			$longitude < -100
+		) {
+			return 'sfo1';
+		} else if (
+			in_array($country, $northWest) ||
+			$longitude >= -100
+		) {
+			return 'nyc3';
+		} else {
+			return array('nyc3', 'sfo1');
+		}
+
+	// Europe
+	} else if ( $continent == 'EU' ) {
+		// These lists are based on who is connected to which international exchange directly.
+		// They are by no means exclusive.
+		$isles = array('GB', 'IM', 'IE', 'FO', 'IS', 'GG', 'JE', 'GI');
+		$germanic = array('DE', 'PT', 'ES', 'FR', 'AD', 'BE', 'LU', 'IT', 'AT', 'CZ', 'PL');
+		$vikings = array('NL', 'SX', 'DK', 'NO', 'SE', 'FI', 'SJ');
+		// Great Britain
+		if ( in_array($country, $isles) ) {
+			return 'lon1';
+		// Germanics
+		} else if ( in_array($country, $germanic) ) {
+			return 'fra1';
+		// Vikings
+		} else if ( in_array($country, $vikings) ) {
+			return 'ams3';
+		// Everywhere else
+		} else {
+			return array('ams3', 'fra1');
+		}
+
+	// South America
+	} else if ( $continent == 'SA' ) {
+		return array('nyc3', 'sfo1');
+
+	// Africe
+	} else if ( $continent == 'AF' ) {
+		return array('fra1', 'ams3');
+
+	} else if (
+		// Asia
+		$continent == 'AS' ||
+		// Oceania
+		$continent == 'OC' ||
+		// Antarctica
+		$continent == 'AN'
+	) {
+		return 'sgp1';
+
+	// Other
+	} else {
+		return array('nyc3', 'ams3');
+	}
+
+}
+
+function ipHash($hostname, $debug = false) {
```

Comment:
What does this function do? Please add comments and explain it a bit, because I have no clue frm reading this code